### PR TITLE
Throws exception on listing job executions with invalid batch job.

### LINKED ISSUE #1822: Dataflow Tasks are not working with Spring Batch
I'm having Spring Batch ```job``` that is also dataflow ```task```. When I run this job everything seems OK, In ```Tasks``` > ```Executions``` I can see that tasks finished successfully. On the other hand when I go to ```Jobs``` tabs I'm getting this error (in command line):

    java.lang.NullPointerException: null
        at org.springframework.cloud.dataflow.server.service.impl.DefaultTaskJobService.getTaskJobExecution(DefaultTaskJobService.java:240) ~[spring-cloud-dataflow-server-core-1.2.2.RELEASE.jar!/:1.2.2.RELEASE]
        at org.springframework.cloud.dataflow.server.service.impl.DefaultTaskJobService.getTaskJobExecutionsForList(DefaultTaskJobService.java:233) ~[spring-cloud-dataflow-server-core-1.2.2.RELEASE.jar!/:1.2.2.RELEASE]
        at org.springframework.cloud.dataflow.server.service.impl.DefaultTaskJobService.listJobExecutions(DefaultTaskJobService.java:103) ~[spring-cloud-dataflow-server-core-1.2.2.RELEASE.jar!/:1.2.2.RELEASE]
        at org.springframework.cloud.dataflow.server.controller.JobExecutionController.list(JobExecutionController.java:91) ~[spring-cloud-dataflow-server-core-1.2.2.RELEASE.jar!/:1.2.2.RELEASE]
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[na:1.8.0_112]
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) ~[na:1.8.0_112]
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[na:1.8.0_112]

Spring Dataflow version = ```1.2.2```

### PR DESCRIPTION
**Pull request by cppwfs**
Throws exception if user executes job execution list when one or more of the batch jobs were not a Spring Cloud Task

resolves #1822

### CODE REVIEW

**Comment by cppwfs** [comment]
Rebased.

File: spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/controller/RestControllerAdvice.java [hunk 1]
```diff
@@ -34,6 +34,7 @@
 import org.springframework.cloud.dataflow.server.repository.DuplicateStreamDefinitionException;
```
Comment (chrisjs) [inline spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/controller/RestControllerAdvice.java]: update copyright date

File: spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/repository/NoSuchTaskBatchException.java [hunk 2]
```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright 2017 the original author or authors.
```
Comment (chrisjs) [inline spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/repository/NoSuchTaskBatchException.java]: update copyright date

File: spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/service/impl/DefaultTaskJobService.java [hunk 3]
```diff
@@ -36,6 +36,7 @@
 import org.springframework.cloud.dataflow.rest.job.TaskJobExecution;
```
Comment (chrisjs) [inline spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/service/impl/DefaultTaskJobService.java]: update copyright date

File: spring-cloud-dataflow-server-core/src/test/java/org/springframework/cloud/dataflow/server/controller/JobExecutionControllerTests.java [hunk 4]
```diff
@@ -61,7 +61,7 @@
 @RunWith(SpringRunner.class)
```
Comment (chrisjs) [inline spring-cloud-dataflow-server-core/src/test/java/org/springframework/cloud/dataflow/server/controller/JobExecutionControllerTests.java]: update copyright date

**Comment by chrisjs** [comment]
merged to master and back ported to 1.3.x

**Pull request closed without merging**